docs: Add resolved issue to release notes (#5260)
hashicorp/boundary: website/content/docs/release-notes/v0_18_0.mdx

    </td>
  </tr>

  <tr>
    <td style={{verticalAlign: 'middle'}}>
    0.18.0 (Fixed in 0.18.1)
    </td>
    <td style={{verticalAlign: 'middle'}}>
    Users are incorrectly removed from managed groups
    </td>
    <td style={{verticalAlign: 'middle'}}>
    If your organization has over 10,000 managed groups, Boundary may incorrectly remove users from the managed group memberships.
    <br /><br />

>     In version 0.18.0 and earlier, there was a maximum number of managed groups allowed for an auth method. If you had over 10,000 managed groups, Boundary may have incorrectly removed users from the groups. This issue is fixed in version 0.18.1. There is no longer a maximum number of managed groups.

Reviewer: I think "allowed" might imply that this was prohibited, when it was just silently unsupported. Also, it might help to explain how this happened. Maybe
```suggestion
    In version 0.18.0 and earlier, there was a maximum number of managed groups supported for an auth method. If you had over 10,000 managed groups, Boundary may have incorrectly removed a user from a group when authenticating. This issue is fixed in version 0.18.1. There is no longer a maximum number of managed groups.
```

Author: Yep, I think that is a helpful addition. Thank you Johan!!
